app/webpacker/components/Panel/pages/RunValidatorsPage/CompetitionRangeSelector.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Form } from 'semantic-ui-react';
+import { QueryClient, useQuery } from '@tanstack/react-query';
+import UtcDatePicker from '../../../wca/UtcDatePicker';
+import getCompetitionCount from './api/getCompetitionCount';
+import Loading from '../../../Requests/Loading';
+import Errored from '../../../Requests/Errored';
+
+const RUN_VALIDATORS_QUERY_CLIENT = new QueryClient();
+
+export default function CompetitionRangeSelector({ range, setRange }) {
```

Comment:
Why are you passing in `range` instead of just passing the dates separately?

Reply:
If you want to maintain it as two separate states anyways, then that feels like a strong argument for passing them in as individual states as well.

> Hence I thought of passing the range as it is.
I don't understand the "hence", because it implies a logical connection. But if your original intention is to maintain one state each, then the logical thing would also be to pass one state each. (And that's exactly what I suggested in my original comment as well). Am I missing something?

Reply:
As per our chat in slack, I've created `setStartDate` and `setEndDate` methods and used that instead of inline logic.

Reply:
I haven't seen that slack discussion, but I don't understand the point of keeping the dates in a single range object/state. The 2 dates are passed individually to the back-end, and in the front-end the 2 dates are always updated separately (as far as I can see, correct me if I'm wrong). It's extra effort/code. It's quite minor though.

Reply:
Sorry for not explaining it in detail here. In `RunValidatorsForm`, I'm having two ways of selecting competitions: through competition ID and through range. For each of these ways, I'm maintaining **one** state - `selectedCompetitionIds` and `selectedCompetitionRange`.

If I need to split `startDate` and `endDate`, I have to split them in `RunValidatorsForm` and I personally like not to split `selectedCompetitionRange` because in that class I want to consider that as a single state.

It's not like I won't split in `RunValidatorsForm`, but I prefer not to split. If you or Gregor still ask me to split, I'll definitely go ahead and split it.